torch/fx/experimental/symbolic_shapes.py

```diff
@@ -648,6 +648,53 @@ def check_supported_type(x):
                 sympy.simplify(base / gcd), sympy.simplify(divisor / gcd)
             )
 
+class Mod(sympy.Function):
+    """
+    Mod(a, b) => a % b
+    """
+
+    nargs = (2,)
+    is_integer = True
+    precedence = 50
+
+    def _sympystr(self, printer):
+        base = printer.parenthesize(self.args[0], self.precedence)
+        divisor = printer.parenthesize(self.args[1], self.precedence)
+        return f"{base}%{divisor}"
+
+    @classmethod
+    def eval(cls, base, modulus):
+        if base == 0 or modulus == 1:
+            return sympy.Integer(0)
+
+        if (
+            isinstance(base, sympy.Integer)
+            and isinstance(modulus, sympy.Integer)
+        ):
+            return base % modulus
+
+        if isinstance(base, sympy.Add):
+            new_terms = []
+            all_positive = True
+            for term in base.args:
+                if sympy.gcd(term, modulus) != modulus:
+                    if (isinstance(term, sympy.Integer) and term < 0) or (
+                        isinstance(term, sympy.Mul)
+                        and isinstance(term.args[0], sympy.Integer)
+                        and term.args[0] < 0
```

Comment:
This condition looks wrong. Why can't we use sympy assumptions to determine if the term is negative or not?

Reply:
This was just mimicked from the existing ModularIndexing code. cc: @ngimel

Reply:
I'm not against using sympy assymptions, I think when I just tried to put comparison in ModularIndexing it errored out, but I don't remember. 
This exemption shouldn't be necessary really, in ModularIndexing it exists only to work around triton bug. (e.g. with valid simplifications (x - y) % x will turn in to -y % x and iirc triton has problems with negative FloorDivs or modulos, but we probably don't need to care about it in symbolic_shapes